bin/release.sh

```diff
@@ -79,9 +92,51 @@ update_version() {
         fi
     else
         status "Updating" "$crate from $curr_version to $version"
-        sed -i \
-            "/\[package\]/,/\[.*dependencies\]/{s/^version = \"$curr_version\"/version = \"$version\"/}" \
-            "$cargo_toml"
+        cargo set-version -p $crate $version
+    fi
+
+    # If we're releasing console-api, we need to update its version in
+    # the other crates in the workspace too.
+    if [[ "$crate" == "console-api" ]]; then
+        local cargo_upgrade=(cargo upgrade --offline -p console-api@$version)
+        if [[ "$verbose" ]]; then
+            cargo_upgrade+=("$verbose")
+        fi
+        
+        "${cargo_upgrade[@]}"
+        files+=($(ls Cargo.lock */Cargo.toml))
+    fi
+}
+
+commit() {
+    if ! [[ -z "${1+1}" ]] && [[ "$1" == "--amend" ]]; then
+        status "Amending" "release commit"
+        amend="$1"
+    else
+        status "Creating" "release commit"
+        amend=""
+    fi
+
+    # Prepare a commit message including the changelog from just this release.
+    tmp_changelog_path="tmp-changelog"
+    tmp_commit_msg_path="tmp-commit-msg"
```

Comment:
hmm, this is a bit unfortunate, since it creates the temporary files in the current working dir, and if the script exits early due to a command failing, the temporary files will be left behind.

we should maybe consider using `mktmp` to create a temporary dir files in the system's temporary directory (e.g. `/tmp`), instead. that way, even if we're not able to properly clean up our tempfiles, the system will eventually delete them, and we won't have issues with a pre-existing temp file with the same name.

see [here](https://unix.stackexchange.com/a/84980) for details on how to use `mktemp` in a cross platform (linux & BSDs like MacOS) way.

Reply:
Done.